content/php/concepts/string-functions/terms/htmlspecialchars/htmlspecialchars.md

```diff
@@ -0,0 +1,102 @@
+Subjects:
+  - 'Computer Science'
+  - 'Web Design'
+  - 'Web Development'
+Tags:
+  - 'Strings'
+  - 'Functions'
+CatalogContent: 
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+The **`htmlspecialchars()`** function takes a string of special characters and converts the string to HTML entities. The special characters have special significance in HTML and need to be represented by HTML entities to preserve their meaning. The special characters and their converted HTML entities are listed in the table below:
+
+| Character | Converted HTML entity |
+|-----------|-----------------------|
+|&          | `&amp;`               |                       
+|"          | `&quot;`              |         
+|'          | `&#039;`              |
+|<          | `&lt;`                |
+|>          | `&gt;`                |
+
+## Syntax
+
+```pseudo
+htmlspecialchars(string $string, int $flag, $string $encoding =null, bool $double_encode = true)
+```
+
+The `htmlspecialchars()` takes four parameters.
+
+- `$string` : A required parameter that specifies the string to be converted.
+- `$flag` : An optional string value that specifies how to handle quotes.
+
+The available flags are listed in the table below:
+
+| Constant       | Description                                                                                                          |
+|----------------|----------------------------------------------------------------------------------------------------------------------|
+|`ENT_COMPAT`    | The default frag. it encodes only double quotes                                                                     |
+|`ENT_QUOTES`    | Encodes both double and single quotes                                                                                |
+|`ENT_NOQUOTES`  | Does not encode double nor single quotes                                                                             |
+|`ENT_IGNORE`    | Discards invalid encoding instead of the function returning an empty string                                          |
+|`ENT_SUBSTITUTE`| Replaces invalid encoding with a unicode character  U+FFFD (UTF-8) or &#xFFFD; instead of returning an empty string  |
+|`ENT_DISALLOWED`| Replaces invalid code for the specified doctype with a Unicode Replacement Character U+FFFD (UTF-8) or &#FFFD;       |
+|`ENT_HTML401`   | Handles code as HTML 4.01                                                                                            |
+|`ENT_HTML5`     | Handles code as HTML 5                                                                                               |
+|`ENT_XML1`      | Handles code as XML 1                                                                                                |
+|`ENT_XHTML1`    | Handles code as XHTML                                                                                                |
+
+- `$encoding`: An optional string value that specifies the character-set to use when converting characters. 
```

Comment:
I think it's cleaner if all of the parameters are together instead of being separated by the tables